Chrysalis-Papilio/Soen-490-Papilio-Backend
# PAPILIO-108-task

## General

The add employee flows does not account for already created firebase user. The backend requires a user to have a firebase_id when being inserted in our database. To make sure we work with the backend, we will rewrite the flow to have the correct information.

We will provide all the information about the employee in the company dashboard. Then the employee account will be created in firebase and the credentials will then be passed towards the backend with the information we require.

_Issue number_: Papilio-108

## Task description
- Add protection to destroy the employee in firebase when there is a problem in the backend and the employee end up not being created in our backend.

## Manual Test
Make sure that you have the appropriate variables in the backend .env file. (Contact me if you don't)

1. start the backend
`npm run build && npm start`
2. In postman, create a new employee for a company `/api/business/addEmployee/:businessId`
3. In Firebase, create an firebase_id for that employee
4. Send the same query once again with the same firebase_id
5. You should see an error message
6. In Firebase, check that the employee account has been deleted

This is not a situation that would happened in real life since firebase make sure that no 2 user have the same email but it works to show the use case. This is more an insurance if something is wrong and the user passes all our checks.

## Testing
To run all the unit tests, run the following code: npm test
To make sure that the tests have reach the appropriate level of coverage, run the following code: npm run coverage
